library/std/src/sys/pal/unix/process/process_common.rs

```diff
@@ -580,8 +580,49 @@ impl fmt::Debug for Command {
 
             debug_command.finish()
         } else {
+            /// Only escape arguments when necessary, otherwise output it
+            /// unescaped to make the output easier to read.
+            fn relaxed_escaping(bytes: &[u8], is_arg: bool) -> impl fmt::Display + use<'_> {
+                // Don't escape if all the characters are likely
+                // to not be a problem when copied to the shell.
+                let can_print_safely = move |&c| {
+                    // See: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_02
+                    match c {
+                        // The documentation linked above says that `=`:
+                        // > may need to be quoted under certain circumstances
+                        //
+                        // Because they may be parsed as a variable assignment.
+                        // But in argument position to a command, they
+                        // shouldn't need escaping.
+                        b'=' if is_arg => true,
+                        // As per documentation linked above:
+                        // > The application shall quote the following characters
+                        b'|' | b'&' | b';' | b'<' | b'>' | b'(' | b')' | b'$' | b'`' | b'\\'
+                        | b'"' | b'\'' | b' ' | b'\t' | b'\n' => false,
```

Comment:
Could we get a test for the quoting of `"`, please?

Also, seconding the premise that we need to quote `!`.

Since this is UNIX-specific, I think it'd be reasonable to always use single-quotes for quoting, and just handle `'` specially.

Reply:
> Could we get a test for the quoting of `"`, please?

There is a test for that? In [library/std/src/sys/pal/unix/process/process_common/tests.rs](https://github.com/rust-lang/rust/pull/132484/files#diff-f89223c7b7312c02e6ead1265c6125d0f6a8fe3a0f0d6a510760198a23d0f205):
```
check("a\"", r#""a\"""#);
```

Or perhaps I'm misunderstanding?